### PR TITLE
Enhancement/4620 all admin ownership

### PR DESCRIPTION
**Pull request by asvinb**
## Summary

Addresses issue:

- #4620 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: includes/Core/Modules/Modules.php [hunk 1]
```diff
@@ -325,6 +325,52 @@ function ( $data ) {
...
+												break;
+											}
+										} else {
+											$changed_settings = true;
+											break;
+										}
+									} elseif ( $old_values[ $module_slug ][ $setting_key ] !== $setting_value ) {
+										$changed_settings = true;
+										break;
+									}
```
Comment (tofumatt) [inline includes/Core/Modules/Modules.php:356]: I'm finding the logic here just a bit tough to follow 😅 

I got it eventually, but could you add some comments outlining the conditions to consider things changed, as these nested if/elses are tough to scan at a glance.

**Review by tofumatt — CHANGES_REQUESTED** [review]
This looks great, but can you add a few comments to clarify the conditions for changes? I think it'll just help the next dev understand it at a glance.

Then this should be good-to-go, although it seems like there are more conditions to trigger a "change" as having happened than tests cases, so if you wanted to add a few more I wouldn't say no 😉

File: includes/Core/Modules/Modules.php (hunk 1 above)
Comment (asvinb) [reply includes/Core/Modules/Modules.php]: Sounds good. Can you check again @tofumatt ?

**Review by tofumatt — APPROVED** [review]
That's awesome, thanks! 👍🏻

**Review by felixarntz — COMMENTED** [review]
@asvinb @tofumatt Let's avoid in the future to put massive functions like this as closures. Class methods are more suitable for most of this.

**Pull request merged**